arcgis-ios-sdk-samples/Augmented reality/View hidden infrastructure in AR/ViewHiddenInfrastructureARPipePlacer.swift

```diff
@@ -0,0 +1,224 @@
+                elevation.intValue <= 10 else { return }
+            // Pass back the elevation value.
+            completion(elevation)
+        }
+        alert.addAction(cancelAction)
+        alert.addAction(doneAction)
+        alert.preferredAction = doneAction
+        present(alert, animated: true)
+    }
+    
+    // MARK: UIViewController
+    
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        if segue.identifier == "showViewer",
+            let controller = segue.destination as? ViewHiddenInfrastructureARViewer,
+            let graphics = pipeGraphicsOverlay.graphics as? [AGSGraphic] {
+            controller.pipeGraphics = graphics
+        }
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // Add the source code button item to the right of navigation bar.
+        (navigationItem.rightBarButtonItem as? SourceCodeBarButtonItem)?.filenames = [
+            "ViewHiddenInfrastructureARPipePlacer",
+            "ViewHiddenInfrastructureARViewer",
+            "ViewHiddenInfrastructureARCalibrationViewController"
+        ]
+        
+        // Configure the elevation surface used to place drawn graphics relative to the ground.
+        elevationSurface.elevationSources.append(elevationSource)
+        elevationSource.load { [weak self] error in
+            guard let self = self else { return }
+            if let error = error {
+                self.presentAlert(error: error)
+            }
+        }
+        
+        // Add a KVO to update the button states.
+        graphicsObservation = pipeGraphicsOverlay.observe(\.graphics, options: .initial) { [weak self] overlay, _ in
+            guard let self = self else { return }
+            // 'NSMutableArray' has no member 'isEmpty'; check its count instead.
+            let graphicsCount = overlay.graphics.count
+            let hasGraphics = graphicsCount > 0
+            self.trashBarButtonItem.isEnabled = hasGraphics
+            self.cameraBarButtonItem.isEnabled = hasGraphics
+        }
+        
+        // Set location display.
+        setStatus(message: "Adjusting to your current location…")
+        locationDataSource.locationChangeHandlerDelegate = self
+        locationDataSource.start { [weak self] error in
+            guard let self = self else { return }
+            if let error = error {
+                self.presentAlert(error: error)
+            }
+        }
+    }
+}
+
+// MARK: - Location change handler
+
+extension ViewHiddenInfrastructureARPipePlacer: AGSLocationChangeHandlerDelegate {
+    func locationDataSource(_ locationDataSource: AGSLocationDataSource, locationDidChange location: AGSLocation) {
+        let newViewpoint = AGSViewpoint(center: location.position!, scale: 1000)
+        mapView.setViewpoint(newViewpoint, completion: nil)
+        // Stop auto-adjusting when it is accurate enough.
+        if location.horizontalAccuracy < 20 {
```

Comment:
This prevents the toolbar buttons from being enabled. Please remove or adjust this